### ISSUE TITLE
Framework: reduce size of MPI_RELEASE_DEBUG_SHARED_PT build

### ISSUE THREAD

**Issue by ibaned** [trilinos/Trilinos#1404]
So, I tried to do a `checkin-test-sems` run for #1396 and ran out of space on CEE.

The build directory is 70 GB (at the point it ran out of memory), and my entire home directory quota is 100 GB.

Looking around, it seems like a lot of space could be saved by not building examples. Does this make sense and is there a way to do it?

@bartlettroscoe

**Comment by ibaned** [comment]
hmm something went wrong in my calculations... different tools are reporting different size of things. my question about examples stands, but may not be whats really consuming the most memory.

**Comment by ibaned** [comment]
nevermind, it just took a while for the `df` command to realize files had been removed. I think the original top comment is accurate.

**Comment by bartlettroscoe** [comment]
@ibaned, don't build in your CEE LAN home directory.   That is a super slow NFS-mounted drive.  Instead, use local scratch space.  See:

* https://snl-wiki.sandia.gov/display/TRIL/Development+and+Testing+on+CEE+LAN+Machines

If you don't have access to a local /scratch drive, then follow the instructions at:

* https://snl-wiki.sandia.gov/display/TRIL/Development+and+Testing+on+CEE+LAN+Machines#DevelopmentandTestingonCEELANMachines-Overview

to get access to a "physical" CEE LAN machine with a fast /scratch drive.

Trilinos is huge.  I just recently built all of the PT packages and tests and here is the size of the build directory:

```
$ du -sh * | sort -rh | less
72G     MPI_RELEASE_DEBUG_SHARED_PT
```

And that is with shared libraries!  My guess is that most of this bloat is due to a lot of duplicate implicit template instantiation.  To fix that, Trilinos developers need to do #1388.  I will add some other stories that will help with that as well.

As for disabling examples, that is not a good idea.  We need to keep the examples working along with all of the tests.

Please contact me offline if you have questions about how to get access to a CEE LAN machine with a fast large /scratch disk.

**Comment by ibaned** [comment]
This question was answered in full offline. Trying the new system now...
